### PR TITLE
[NFC][ntuple] add v6.34 release notes

### PR DESCRIPTION
**Pull request by jblomer**

### CODE REVIEW

**Review by dpiparo — APPROVED** [review]

File: README/ReleaseNotes/v634/index.md [hunk 1]
```diff
@@ -96,6 +96,26 @@ The following interfaces are deprecated and will be removed in future releases:
...
+  RNTuple data written as of this release. Please note that this version breaks compatibility with experimental RNTuple
+  data written with releases up to v6.34. Please also note that the RNTuple API is not yet moving out of
+  `ROOT::Experimental`.
+* Support for low-precision on-disk floating point representation. This can be enabled through
+  `RField<float|double>::SetTruncated()` (truncated mantissa) and `RField<float|double>::SetQuantized()`
+  (scaled integer representation).
+* Link RNTuple self-description to the common ROOT streamer infrastructure. As a result, `TFile::MakeProject()`
+  properly creates header files for classes used in RNTuple data.
+* First version of the new `RNTupleProcessor` class. The `RNTupleProcessor` supports iteration of joined RNTuple
+  data sets (à la friends and chains).
```
Comment (enirolf) [inline README/ReleaseNotes/v634/index.md:111]: This release only supports chains for now, I think we should clarify that.

Comment (enirolf) [reply README/ReleaseNotes/v634/index.md:111]: My suggested change: 
```suggestion
* First version of the new `RNTupleProcessor` class. The `RNTupleProcessor` will support iteration of composed RNTuple data sets (comparable to and improving upon TTree friends and chains). This release supports chained (vertically composed) RNTuples. Other types of concatenations will be added in subsequent releases.
```

File: README/ReleaseNotes/v634/index.md [hunk 2]
```diff
@@ -96,6 +96,25 @@ The following interfaces are deprecated and will be removed in future releases:
 
 ## I/O Libraries
 
+## RNTuple Libraries
+
+* The first version of the on-disk binary format is finalized. Future versions of ROOT will be able to read back
```
Comment (pcanal) [inline README/ReleaseNotes/v634/index.md:101]:
```suggestion
* The first version of the `RNTuple` on-disk binary format is finalized. Future versions of ROOT will be able to read back
```
let's be pedantic to allow easy accurate copy/paste :)

File: README/ReleaseNotes/v634/index.md [hunk 3]
```diff
@@ -96,6 +96,25 @@ The following interfaces are deprecated and will be removed in future releases:
...
+  (scaled integer representation).
+* Link RNTuple self-description to the common ROOT streamer infrastructure. As a result, `TFile::MakeProject()`
+  properly creates header files for classes used in RNTuple data.
+* First version of the new `RNTupleProcessor` class. The `RNTupleProcessor` will support iteration of composed RNTuple data sets (comparable to and improving upon TTree friends and chains). This release supports chained (vertically composed) RNTuples. Other types of concatenations will be added in subsequent releases.
+* Support for cluster staging in the `RNTupleParallelWriter`. Cluster staging enables users to enforce a certain
+  logical cluster ordering in the presence of parallel cluster writing.
+* Support for Direct I/O for writing. This gives access to the peak performance of modern NVMe drives.
+* Support for a "streamer field" that can wrap classic ROOT I/O serialized data for RNTuple in cases where native
+  RNTuple support is not possible (e.g., recursive data structures). Use of the streamer field can be enforced
+  through the LinkDef option `rntupleStreamerMode(true)`.
```
Comment (pcanal) [inline README/ReleaseNotes/v634/index.md:116]: Consider adding
```suggestion
  through the LinkDef option `rntupleStreamerMode(true)`.  This features is similar to the unsplit/level-0 split branch in `TTree`
```

**Pull request merged**